backend/src/chats/chat/chat.controller.ts

```diff
@@ -29,15 +40,38 @@ export class ChatController {
 	}
 
 	@Get()
-	findAll() {
-		return this.chatService.findAll({ relations: { has_users: true } });
+	findAll(
+		@Query() userRelationsQuery?: ChatRelationsQueryDto,
+		@Body() userRelationsBody?: ChatRelationsBodyDto,
+	) {
+		const userRelationsDto = Object.keys(userRelationsBody ?? {}).length
+			? userRelationsBody
+			: Object.keys(userRelationsQuery ?? {}).length
+			? userRelationsQuery
+			: this.defaultRelationships;
+
```

Comment:
Vincent, I understand what the result of this block of code is, but I have no idea how it is achieved.... could solve some of this abstraction for me with a short explanation please?

Reply:
wait i get it, as bunch of ternary expression, if URbody, map and use that, else if URquery map and use that, else default, right?

Reply:
yes

i know, i hate it too.. but i gave up in live at some point...
I wanted to make a helper function that just gets the 3 options as params and figures it out with if/else. 
But I feel like those kind of helpers should live in their own file in TS/nestjs, or at least not be a method of the controller. So I gave up on the thought of figuring out where to put it and what to name it...